.github/workflows/after-push-to-branch.yaml

```diff
@@ -0,0 +1,36 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Run the CD script after pushes to the release branches
+on:
+  push:
+    branches:
+    - "master"
```

Comment:
Good call on the description. I think the intent is to explore more flexible function release workflows and not limit that to the release branch based triggering. So "master" is the match.

Reply:
I think it's a meaningful difference whether the intent is for release branches or `master`. For `master` I would expect this to build artifacts for all functions, whereas for release branches I might expect this to only build artifacts for one function (given that the release branches are per-function).

Reply:
Do you know if we have a doc about "release branches are per-function" and more broadly the branching/tagging strategy?

Branches per function isn't something I've seen in the wild in other projects; rather I think that when we have a "big headline release" that we offer long-term support for, we cut a release branch for that, and then we backport security fixes or bad-enough-bugfixes from main to it/them.  But I would imagine that most function releases would be done as tags, and I was trying to follow the pattern of kpt packages here.  But really I'm just trying to figure out what makes sense, what we should recommend to people in the ecosystem that want to maintain their own function repos.

Reply:
I'm not familiar with whether there is a doc describing the process, I'm mostly going off of context clues from existing branches/tags. My recollection is a bit fuzzy and may be incorrect but this is what I remember from the tribal knowledge passed to me when I started working on the project.

I agree it's a bit atypical but the current approach seems to be a sort of multi-tenant/mono-repo approach where each function is treated as a separate product with it's own release cycle/cadence. Release branches and tags both seem to be prefixed with the associated function that is being released.

I do think it could be possible to transition to an approach where we release all functions as a single release, although my understanding is that would be a deviation from the current approach.

Reply:
Right, I'm thinking we maintain the current release process in that "real releases" of individual functions are tagged.   If people want to use release branches (whether per function or not), they can do so.

What we are likely adding is continuous builds of the release branches.  I see those as more useful for development; if people are actually going to incorporate them into their packages they should probably do a tagged release.  (But there is another alternative, which is that we can "promote" a particular version, this is what `registry.k8s.io` does: https://github.com/kubernetes/k8s.io/tree/main/k8s.gcr.io )

Reply:
So we've dug in and I don't think we need the branch-per-function (though others may disagree - please comment if so :-) ).  We're essentially building a parallel (hopefully simpler) build process here, so we can also iterate and discover if we do need these branches.  As I see it, on the master branch (and any release branches, though we don't have any generic release branches yet) we build all the functions and push them with a SHA-derived tag.   When a function is tagged, we push only that function.  The format of these tags is `<directory>/<tag>`  (There's some ambiguity about what it would mean to push a top-level tag of v0.0.1, for example, but in the current pattern that would be a function that lives in the root, which is valid in a repo-per-function approach).

If this sounds good I can write this up in a markdown file somewhere in this repo.

I renamed the description to be 'Run the CD script after pushes to branches", although for the moment I'm restricting it to the master branch, so I don't interfere with anything that might be happening in terms of per-function branches.